### PR TITLE
Feature/11 navbar component

### PR DESCRIPTION
**Pull request by te4-gabriel-stark**
## What issue did you implement or fix?
Implemented the navbar according to #11, also fixed bug #35.

## Summary
- Added a navbar that when you're not logged in you have a login button that takes you to the login page.
- When you are logged in it has a logout button.
- Also added a loading bar to the LoginFormComponent.
- You now stay logged in on refresh.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## How can this implementation be improved?
- Add more links later when those pages exist.

### CODE REVIEW

File: frontend/src/views/Home.vue [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
-  name: "home",
+
+@Component({
   components: {
-    HelloWorld
   }
-};
+})
+export default class Home extends Vue {
+  private userModule = UserModule;
```
Comment (TE4-jacob-klaren) [inline frontend/src/views/Home.vue]: hmmmm

File: frontend/src/store/modules/UserModule.ts [hunk 2]
```diff
@@ -42,10 +42,10 @@ class UserModule extends VuexModule {
...
     @Action
-    login({email, password} : ILoginPayload) {
+    async login({email, password} : ILoginPayload) {
         this.setIsLoading(true);
         this.setFailure("");
-        axios.post("http://localhost:9292/api/v1/users/login", {email, password})
+        await axios.post("http://localhost:9292/api/v1/users/login", {email, password})
             .then( response => {
                 const decoded : IUser = jwt.decode(response.data.token) as IUser;
                 decoded.username = response.data.username
```
Comment (TE4-Daniel-Kull) [inline frontend/src/store/modules/UserModule.ts]: I quote you on dis "My code follows the style guidelines of this project" 
But wait. where is da semi colon ∂Ω∂œı√ƒéıœƒπçœı√÷ﬁß∂ç‹˛›éΩœıƒ˛Ωœıƒüé˛œΩ[ƒyzqo
Illuminati confirmed:eyes::eyes::eyes::eyes::eyes::eyes::eyes:
```suggestion
                decoded.username = response.data.username;
```

File: frontend/src/views/Home.vue (hunk 1 above)
Comment (te4-gabriel-stark) [reply frontend/src/views/Home.vue]: Om man inte gör så kan man inte använda userModule i template, det är ganska retarded tbh.

Comment (TE4-jacob-klaren) [reply frontend/src/views/Home.vue]: 🧐

Comment (te4-gabriel-stark) [reply frontend/src/views/Home.vue]: Om man inte gör så kan man inte använda userModule i template, it's hella dumb.

File: frontend/src/components/NavbarComponent.vue [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<template>
+    <div>
+        <v-toolbar dense>
+            <v-toolbar-title>Log-O-Matic</v-toolbar-title>
+            <v-spacer></v-spacer>
+            <v-toolbar-items>
+                <v-btn to="/login" text v-if="!userModule.currentUser.username">Login</v-btn>
+                <v-btn @click="userModule.logout()" text v-else>Logout</v-btn>
```
Comment (TE4-Daniel-Kull) [inline frontend/src/components/NavbarComponent.vue]: #11 Did we agree on not showing the name of the current user? I seriously don't remember plz help. It's not important, but if you feel like it, you can add a display for the current username. Eh, just push it anyway I guess.

Comment (te4-gabriel-stark) [reply frontend/src/components/NavbarComponent.vue]: Kollar man i DOD så står det med, men kollar man bilderna så är det inte med. Så jag följde bilden :)))))))))

File: frontend/src/store/modules/UserModule.ts (hunk 2 above)
Comment (te4-gabriel-stark) [reply frontend/src/store/modules/UserModule.ts]: På axios saken så forstätter tekniskt sett raden efter, hence inget ;

**Comment by TE4-emanuel-strom** [comment]
+1

File: frontend/src/components/NavbarComponent.vue (hunk 3 above)
Comment (TE4-Daniel-Kull) [reply frontend/src/components/NavbarComponent.vue]: okidokes! :D

**Pull request merged**